staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -144,6 +145,10 @@ func validateCustomResourceDefinitionSpec(spec *apiextensions.CustomResourceDefi
 			}
 		}
 	}
+	if spec.PreserveUnknownFields != nil && *spec.PreserveUnknownFields == true {
```

Comment:
I'd actually flip this around the other way... if someone wants to use defaults in their schema but hasn't enabled pruning, I'd rather use that as an enticement to enable pruning than simply reject the default values

```
if allowDefaults && specHasDefaults(spec) {
  mustBeStructural = true
  if spec.PreserveUnknownFields == nil || *spec.PreserveUnknownFields == true {
    ... error message on spec.preserveUnknownFields field, must be false in order to use defaults in the schema, etc
  }
}
```

Reply:
fixed

Reply:
fixed